pkg/crd/markers/validation.go

```diff
@@ -298,7 +298,10 @@ func (m MinLength) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
 	return nil
 }
 func (m Pattern) ApplyToSchema(schema *apiext.JSONSchemaProps) error {
-	if schema.Type != "string" {
+	// Allow string types or IntOrStrings. An IntOrString will still
+	// apply the pattern validation when a string is detected, the pattern
+	// will not apply to ints though.
+	if schema.Type != "string" || schema.XIntOrString {
 		return fmt.Errorf("must apply pattern to a string")
```

Comment:
Do we need to update the error message?

Reply:
Can do, I wasn't sure originally though because I couldn't think of a good way to express the `IntOrString`, perhaps `must apply pattern to a string or IntOrString` but I don't like the inconsistency of casing 🤔

Reply:
Have pushed an update with the new error message